services/skus/datastore.go

```diff
@@ -950,24 +950,33 @@ type TimeAwareSubIssuedCreds struct {
 	RequestID    string                    `json:"-" db:"request_id"`
 }
 
-func (pg *Postgres) AreTimeLimitedV2CredsSubmitted(ctx context.Context, blindedCreds ...string) (bool, error) {
+func (pg *Postgres) AreTimeLimitedV2CredsSubmitted(ctx context.Context, requestID uuid.UUID, blindedCreds ...string) (bool, bool, error) {
 	if len(blindedCreds) < 1 {
-		return false, errors.New("invalid parameter to tlv2 creds signed")
+		return false, false, errors.New("invalid parameter to tlv2 creds signed")
 	}
 
 	query := `
 		select exists(
 			select 1 from time_limited_v2_order_creds where blinded_creds->>0 = $1
-		)
+		) as already_submitted,
+		exists(
+			select 1 from time_limited_v2_order_creds where blinded_creds->>0 = $1 and request_id != $2
+		) as mismatch
 	`
 
-	var alreadySubmitted bool
-	err := pg.RawDB().Get(&alreadySubmitted, query, blindedCreds[0])
+	type data struct {
```

Comment:
suggestion(non-blocking): might be worth just returning this type or using named parameters to avoid get the two bools mixed up.

Reply:
addressed: https://github.com/brave-intl/bat-go/pull/2331/commits/e53a580ef2adb14a06f44f6784f86ba80c9d14ec